pkg/git/repository.go

```diff
@@ -252,6 +257,96 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
 	return r, fs.Create(bootstrapFS), nil
 }
 
+var validateRepoWritePermission = func(ctx context.Context, r *repo) error {
+	cert, err := r.auth.GetCertificate()
+	if err != nil {
+		return fmt.Errorf("failed getting repository certificates: %w", err)
+	}
+
+	err = r.Repository.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: r.progress,
+		CABundle: cert,
+	})
+	if err != nil {
+		if !errors.Is(err, gg.NoErrAlreadyUpToDate) {
+			// means there was already a commit to push, and it failed to push it
+			return fmt.Errorf("failed to initialize repository, failed pushing to repository: %w", err)
+		}
+		// no commit to push, all up to date
+		err = r.PushDummyCommit("Validating write permission to project", ctx)
+
+		if err != nil {
+			return fmt.Errorf("failed to initialize repository, failed pushing to repository: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *repo) PushDummyCommit(commitMsg string, ctx context.Context) error {
+	gitlog, err := r.Repository.Log(&gg.LogOptions{})
+	if err != nil {
+		return fmt.Errorf("failed getting git log: %w", err)
+	}
+
+	commit, err := gitlog.Next()
+	if err != nil {
+		return fmt.Errorf("gailed getting last commit: %w", err)
+	}
+
+	hash := commit.Hash
```

Comment:
If we remove the reset part, do we still need this?

Reply:
no